waffle-compiler/src/compileNativeSolc.ts

```diff
@@ -14,7 +14,7 @@ export function compileNativeSolc(config: Config) {
 }
 
 export function createBuildCommand(config: Config) {
-  const command = 'solc';
+  const command = config.compilerVersion || 'solc';
```

Comment:
Not sure if reusing this field is a good idea, but this was the least invasive change.